core/src/html/text_format.rs

```diff
@@ -894,6 +894,7 @@ impl TextSpan {
 #[collect(require_static)]
 pub struct FormatSpans {
     text: String,
+    visual_text: String,
```

Comment:
My idea is to have different strings representing what is visually displayed, and what the underlying text actually is. This fixed a lot of problems I was having implementing this.